src/toil/jobStores/abstractJobStore.py

```diff
@@ -495,8 +495,26 @@ def getConnectedJobs(jobGraph):
             # Delete the job
             self.delete(jobGraph.jobStoreID)
 
+        jobGraphsReachableFromRoot = {id: getJob(id) for id in reachableFromRoot}
+
```

Comment:
I know, but the logic for the function should be that you can turn off the cache and have the jobs never fully resident in memory. Currently not using the cache is totally stupid.